## **Pathogen Genomics Training Workshop**


Register [here](https://docs.google.com/forms/d/1x4La0OaNaTgOCK6cj2FXeNSSdEa5-HAy6LmtiWyTXAA/edit).

**Contact:** Please email [nicholas.bbosa@mrcuganda.org](nicholas.bbosa@mrcuganda.org) for more information.

## **Training Schedule**

### **Day 1: Introduction to Metagenomics Next Generation Sequencing**

**Instructor(s):** 

| Time  | Session |
| ------------- | ------------- |
|0900 - 1000 |    Arrival & Registration |
|1000 - 1030 |    Opening remarks |
|1030 - 1100 |    Coffee break|
|1100 - 1300 |    Molecular Biology for Bioinformatics |
|1300 - 1400 |    Lunch |
|1400 - 1430 |    Introduction to metagenomics |
|1430 - 1500 |    Applications of metagenomics and NGS file formats|
|1500 - 1530 |    Coffee Break |
|1530 - 1630 |    Next Generation Sequencing  |
|1630 - 1700 |    Recap  |

### **Day 2: Working with Metagenomics: Part I**

**Instructor(s):** 

| Time  | Session |
| ------------- | ------------- |
|0900 - 0930 |    Recap |
|0930 - 1030 |    Virtual machine installation |
|1030 - 1100 |    Coffee break|
|1100 - 1300 |    Introduction to linux |
|1100 - 1300 |    Navigating the linux directory structure - Hands-on |
|1300 - 1400 |    Lunch |
|1400 - 1430 |    Workflow for metagenomics data analysis |
|1430 - 1500 |    QC, Adapter and quality trimming |
|1500 - 1530 |    Coffee Break |
|1530 - 1630 |    Mapping metagenomic reads to reference genomes |
|1630 - 1700 |    Generating consensus sequences from alignment maps  |


### **Day 3: Working with Metagenomics: Part II**

**Instructor(s):** 

| Time  | Session |
| ------------- | ------------- |
|0900 - 0930 |    Recap |
|0930 - 1000 |    De novo assembly of metagenomic reads |
|1000 - 1030 |    Taxonomic identification of assembled contigs |
|1030 - 1100 |    Coffee break|
|1100 - 1300 |    Lab practical (Group I)|
|1300 - 1400 |    Lunch |
|1400 - 1600 |    Lab practical (Group II) |
|1600 - 1630 |    Coffee Break |
|1630 - 1700 |    Boat ride  |
